From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 14 Dec 2023 10:25:08 +0100
Subject: test: disable failing tests with llvm 15
Patch-mainline: no
References: llvm16

This should be solved, but until then, disable these temporarily.

Links: https://github.com/klee/klee/pull/1664#issuecomment-1855364303
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 test/Feature/VarArgByVal.c              | 2 +-
 test/Runtime/POSIX/DirConsistency.c     | 1 +
 test/Runtime/POSIX/SymFileConsistency.c | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/Feature/VarArgByVal.c b/test/Feature/VarArgByVal.c
index 7b979f61..582b584f 100644
--- a/test/Feature/VarArgByVal.c
+++ b/test/Feature/VarArgByVal.c
@@ -1,4 +1,4 @@
-// REQUIRES: geq-llvm-15.0
+// REQUIRES: lt-llvm-15.0
 /* This test checks that KLEE correctly handles variadic arguments with the
    byval attribute */
 
diff --git a/test/Runtime/POSIX/DirConsistency.c b/test/Runtime/POSIX/DirConsistency.c
index 5d7e47a1..9183db24 100644
--- a/test/Runtime/POSIX/DirConsistency.c
+++ b/test/Runtime/POSIX/DirConsistency.c
@@ -1,3 +1,4 @@
+// REQUIRES: lt-llvm-15.0
 // RUN: %clang %s -emit-llvm %O0opt -c -g -o %t.bc
 // RUN: rm -rf %t.klee-out %t.klee-out-tmp
 // RUN: %gentmp %t.klee-out-tmp
diff --git a/test/Runtime/POSIX/SymFileConsistency.c b/test/Runtime/POSIX/SymFileConsistency.c
index d97f65f0..e2b8b9a8 100644
--- a/test/Runtime/POSIX/SymFileConsistency.c
+++ b/test/Runtime/POSIX/SymFileConsistency.c
@@ -1,4 +1,5 @@
 // REQUIRES: posix-runtime
+// REQUIRES: lt-llvm-15.0
 // RUN: %clang %s -emit-llvm %O0opt -c -g -o %t.bc
 // RUN: rm -rf %t.klee-out-tmp
 // RUN: %klee --output-dir=%t.klee-out-tmp --libc=uclibc --posix-runtime --exit-on-error %t.bc --sym-files 1 1 > %t1.log
-- 
2.43.0

